### ISSUE TITLE
TypeError: Cannot read properties of undefined (reading '0'), #153 is not fixed

### ISSUE THREAD

**Issue by Krillle** [ilcato/homebridge-fibaro-home-center#169]
@mkz212 #153 is not fixed. Still same issue with v1.4.1:

```
[12/11/2023, 11:43:24] [FibaroHC] Wohnbereich [131]: 23.0 C
[12/11/2023, 11:43:24] [FibaroHC] Wohnbereich [131]: 0 
[12/11/2023, 11:43:24] [FibaroHC] Temperatur 129.0.1 [131]: 24.4 C
[12/11/2023, 11:43:25] TypeError: Cannot read properties of undefined (reading '0')
    at GetFunctions.getCurrentTemperature (/var/lib/homebridge/node_modules/homebridge-fibaro-home-center/src/getFunctions.ts:197:85)
    at Poller.manageValue (/var/lib/homebridge/node_modules/homebridge-fibaro-home-center/src/pollerupdate.ts:110:34)
    at /var/lib/homebridge/node_modules/homebridge-fibaro-home-center/src/pollerupdate.ts:49:18
    at Array.map (<anonymous>)
    at Poller.poll (/var/lib/homebridge/node_modules/homebridge-fibaro-home-center/src/pollerupdate.ts:47:25)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
[12/11/2023, 11:43:25] Got SIGTERM, shutting down Homebridge...
```

**Device 131 API response:**
```
{
  "id": 131,
  "name": "Temperatur 129.0.1",
  "roomID": 4,
  "type": "com.fibaro.temperatureSensor",
  "baseType": "com.fibaro.multilevelSensor",
  "enabled": true,
  "visible": true,
  "isPlugin": false,
  "parentId": 129,
  "remoteGatewayId": 0,
  "interfaces": [
    "zwave"
  ],
  "properties": {
    "parameters": [
      {
        "id": 1,
        "lastReportedValue": 1,
        "lastSetValue": 1,
        "size": 2,
        "value": 1
      },
      {
        "id": 2,
        "lastReportedValue": 1,
        "lastSetValue": 1,
        "size": 2,
        "value": 1
      },
      {
        "id": 3,
        "lastReportedValue": 0,
        "lastSetValue": 0,
        "size": 2,
        "value": 0
      },
      {
        "id": 4,
        "lastReportedValue": 3,
        "lastSetValue": 3,
        "size": 2,
        "value": 3
      },
      {
        "id": 5,
        "lastReportedValue": 50,
        "lastSetValue": 50,
        "size": 2,
        "value": 50
      },
      {
        "id": 6,
        "lastReportedValue": 400,
        "lastSetValue": 400,
        "size": 2,
        "value": 400
      },
      {
        "id": 7,
        "lastReportedValue": 50,
        "lastSetValue": 50,
        "size": 2,
        "value": 50
      },
      {
        "id": 8,
        "lastReportedValue": 400,
        "lastSetValue": 400,
        "size": 2,
        "value": 400
      },
      {
        "id": 9,
        "lastReportedValue": 0,
        "lastSetValue": 0,
        "size": 2,
        "value": 0
      },
      {
        "id": 10,
        "lastReportedValue": 240,
        "lastSetValue": 240,
        "size": 2,
        "value": 240
      },
      {
        "id": 11,
        "lastReportedValue": 240,
        "lastSetValue": 240,
        "size": 2,
        "value": 240
      },
      {
        "id": 12,
        "lastReportedValue": 2,
        "lastSetValue": 2,
        "size": 2,
        "value": 2
      },
      {
        "id": 13,
        "lastReportedValue": 210,
        "lastSetValue": 210,
        "size": 2,
        "value": 210
      }
    ],
    "pollingTimeSec": 0,
    "zwaveCompany": "Heatit",
    "zwaveInfo": "6,3,83",
    "zwaveVersion": "1.5",
    "categories": "[\"climate\"]",
    "configured": true,
    "dead": "false",
    "deadReason": "",
    "deviceControlType": "0",
    "deviceIcon": "1004",
    "emailNotificationID": "0",
    "emailNotificationType": "0",
    "endPointId": "0",
    "log": "",
    "logTemp": "",
    "manufacturer": "",
    "markAsDead": "true",
    "model": "",
    "nodeId": "32",
    "offset": "0.00",
    "parametersTemplate": "720",
    "productInfo": "1,155,0,1,0,1,1,5",
    "pushNotificationID": "0",
    "pushNotificationType": "0",
    "remoteGatewayId": "0",
    "saveLogs": "true",
    "serialNumber": "h'00000001",
    "showFireAlarm": "true",
    "showFreezeAlarm": "false",
    "smsNotificationID": "0",
    "smsNotificationType": "0",
    "unit": "C",
    "useTemplate": "true",
    "userDescription": "",
    "value": "24.50"
  },
  "actions": {
    "reconfigure": 0
  },
  "created": 1698659718,
  "modified": 1698659718,
  "sortOrder": 174
}
```

**Full log:**
```
[12/11/2023, 11:43:35] [HB Supervisor] Starting Homebridge with extra flags: -I -P /var/lib/homebridge/node_modules --strict-plugin-resolution
[12/11/2023, 11:43:35] [HB Supervisor] Started Homebridge v1.7.0 with PID: 26851
[12/11/2023, 11:43:37] Loaded config.json with 2 accessories and 4 platforms.
[12/11/2023, 11:43:37] Loaded 85 cached accessories from cachedAccessories.
[12/11/2023, 11:43:38] ---
[12/11/2023, 11:43:38] Loaded plugin: homebridge-dummy@0.9.0
[12/11/2023, 11:43:38] Registering accessory 'homebridge-dummy.DummySwitch'
[12/11/2023, 11:43:38] ---
[12/11/2023, 11:43:39] Loaded plugin: homebridge-dyson-pure-cool@2.7.6
[12/11/2023, 11:43:39] Registering platform 'homebridge-dyson-pure-cool.DysonPureCoolPlatform'
[12/11/2023, 11:43:39] ---
[12/11/2023, 11:43:39] Loaded plugin: homebridge-fibaro-home-center@1.4.1
[12/11/2023, 11:43:39] Registering platform 'homebridge-fibaro-home-center.FibaroHC'
[12/11/2023, 11:43:39] ---
[12/11/2023, 11:43:40] Loaded plugin: homebridge-http-switch@0.5.36
[12/11/2023, 11:43:40] Registering accessory 'homebridge-http-switch.HTTP-SWITCH'
[12/11/2023, 11:43:40] ---
[12/11/2023, 11:43:41] Loaded plugin: homebridge-irobot@2.1.16
[12/11/2023, 11:43:41] Registering platform 'homebridge-irobot.iRobotPlatform'
[12/11/2023, 11:43:41] ---
[12/11/2023, 11:43:41] Loading 4 platforms...
[12/11/2023, 11:43:41] [FibaroHC] Initializing FibaroHC platform...
[12/11/2023, 11:43:41] [iRobot] Initializing iRobotPlatform platform...
[12/11/2023, 11:43:41] [DysonPureCoolPlatform] Initializing DysonPureCoolPlatform platform...
[12/11/2023, 11:43:41] Loading 2 accessories...
[12/11/2023, 11:43:41] [Alle weg] Initializing DummySwitch accessory...
[12/11/2023, 11:43:42] [Garagentor] Initializing HTTP-SWITCH accessory...
[12/11/2023, 11:43:42] [Garagentor] Switch successfully configured...
[12/11/2023, 11:43:42] [iRobot] Loading accessory from cache: Roomba
[12/11/2023, 11:43:42] [iRobot] Logging into iRobot...
[12/11/2023, 11:43:45] [iRobot] Configuring roomba: Roomba
[12/11/2023, 11:43:46] [iRobot] Restoring existing accessory from cache: Roomba
[12/11/2023, 11:43:47] [DysonPureCoolPlatform] Device with serial number NN2-EU-KKxxxxA: product type 475
[12/11/2023, 11:43:47] [DysonPureCoolPlatform] Device with serial number VS9-EU-KFxxxxA: product type 438
[12/11/2023, 11:43:47] [DysonPureCoolPlatform] Accessories initialized.
Setup Payload:
X-HM://002xxxxxxxW6T
Enter this code with your HomeKit app on your iOS device to pair with Homebridge:
                       
    ┌────────────┐     
    │ xxx-xx-xxx │     
    └────────────┘     
                       
[12/11/2023, 11:43:48] Homebridge v1.7.0 (HAP v0.11.1) (Homebridge) is running on port 51194.
[12/11/2023, 11:43:48] [iRobot] Succefully connected to roomba Roomba
[12/11/2023, 11:43:49] [FibaroHC] Loading accessories
[12/11/2023, 11:44:01] [FibaroHC] power is not a number. 
[12/11/2023, 11:44:01] [FibaroHC] power is not a number. 
[12/11/2023, 11:44:01] [FibaroHC] power is not a number. 
[12/11/2023, 11:44:01] [FibaroHC] power is not a number. 
[12/11/2023, 11:44:01] [FibaroHC] power is not a number. 
[12/11/2023, 11:44:01] [FibaroHC] power is not a number. 
[12/11/2023, 11:46:23] [FibaroHC] Temperatur 44.0.1 [46]: 24.6 C
[12/11/2023, 11:46:33] [FibaroHC] Temperatur 133.0.1 [135]: 29.2 C
[12/11/2023, 11:47:40] [FibaroHC] Wohnbereich [131]: 23.0 C
[12/11/2023, 11:47:40] [FibaroHC] Wohnbereich [131]: 0 
[12/11/2023, 11:47:40] [FibaroHC] Temperatur 129.0.1 [131]: 24.5 C
[12/11/2023, 11:47:40] TypeError: Cannot read properties of undefined (reading '0')
    at GetFunctions.getCurrentTemperature (/var/lib/homebridge/node_modules/homebridge-fibaro-home-center/src/getFunctions.ts:197:85)
    at Poller.manageValue (/var/lib/homebridge/node_modules/homebridge-fibaro-home-center/src/pollerupdate.ts:110:34)
    at /var/lib/homebridge/node_modules/homebridge-fibaro-home-center/src/pollerupdate.ts:49:18
    at Array.map (<anonymous>)
    at Poller.poll (/var/lib/homebridge/node_modules/homebridge-fibaro-home-center/src/pollerupdate.ts:47:25)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
[12/11/2023, 11:47:40] Got SIGTERM, shutting down Homebridge...
```

**Comment by mkz212** [comment]
@Krillle I don't think it's a 131 device because it displays a value. I think the next one after him has a problem.

@ilcato What do you think?

**Comment by Krillle** [comment]
Would a full API response be helpful?

**Comment by mkz212** [comment]
@Krillle You can give but I don't promise anything. I looked through this code many times and found no error. This probably applies to some device that I do not have.

**Comment by Krillle** [comment]
If I see it right, the crash happens when trying to log an error message:
 ` this.platform.log('There was a problem getting value from: ', `${service.IDs[0]} - Err: ${e}`);`

Not a solution, but wouldn't it be a workaround, to modify the error message not accessing the problematic property:
` this.platform.log('There was a problem getting value from: ', `Err: ${e}`);`

**Comment by mkz212** [comment]
@Krillle One fix - the problem does not appear by displaying the log. The log only states that there was an error. The problem arises because the code cannot correctly obtain the value (and thus also display it).

Of course, turning off individual devices is a method of trial and error to track down what causes the problem.

**Comment by mkz212** [comment]
@ilcato Maybe an interesting solution would be the possibility of excluding devices? For example, in the config, field, where one or several IDs are entered and this devices are disabled?

**Comment by Krillle** [comment]
An even more verbose debugging logging option could be helpful to find the problematic device.

**Comment by Krillle** [comment]
Here is my full API response: https://gist.github.com/Krillle/eeee5f373bbe17b302cb81a45590e26f

**Comment by ilcato** [comment]
> @ilcato Maybe an interesting solution would be the possibility of excluding devices? For example, in the config, field, where one or several IDs are entered and this devices are disabled?

@Krillle @mkz212 two ways to exclude devices from the plugin:

1. use a specific user (not an admin one) and grant access to only the needed devices
2. rename the device you want to exclude with an initial _ character

**Comment by Krillle** [comment]
@ilcato  @mkz212 I had a deeper look into the code. IMHO the crash happens here in the CATCH part, after a first error occurred:

```
...if (service.isHeatingZone) {
      try {    <--- THIS TRY FAILS
        const properties = (await this.platform.fibaroClient.getHeatingZone(IDs[0])).body.properties;
        if (!Object.prototype.hasOwnProperty.call(properties, 'currentTemperature')) {
          this.platform.log('No value for Temperature.', '');
          return;
        }
        characteristic.updateValue(properties.currentTemperature);
      } catch (e) {    <--- THIS CATCH IS EXECUTED
        this.platform.log('There was a problem getting value from: ', `${service.IDs[0]} - Err: ${e}`);
                                                              IDs[0] IS UNDEFINED ---^
        return
      }
    }...
```

So changing this line 197 at least stops the crashing (I confirmed that locally):
`this.platform.log('There was a problem getting value from: ', ´Err: ${e}´);`

Checking where the crashing function is called, I think the problem lies here:
at Poller.manageValue (/var/lib/homebridge/node_modules/homebridge-fibaro-home-center/src/pollerupdate.ts:110:34):
```
getFunction.function.call(this.platform.getFunctions, subscription.characteristic, subscription.service, null, change);
                                                                FUNCTION IS CALLED WITH PARAMETER null ---^
```
at GetFunctions.getCurrentTemperature (/var/lib/homebridge/node_modules/homebridge-fibaro-home-center/src/getFunctions.ts:197:85):
 ```
 async getCurrentTemperature(characteristic, service, IDs, properties) {
                                                       ^--- IDs IS INITIALIZED WITH null
    if (service.isClimateZone) {
      try {
        const properties = (await this.platform.fibaroClient.getClimateZone(IDs[0])).body.properties;
                                           SO THIS IDs[0] MUST TROW AN ERROR ---^
        if (!Object.prototype.hasOwnProperty.call(properties, 'currentTemperatureHeating')) {
          this.platform.log('No value for Temperature.', '');
          return;
        }
        characteristic.updateValue(properties.currentTemperatureHeating);
      } catch (e) {
        this.platform.log('There was a problem getting value from: ', `${service.IDs[0]} - Err: ${e}`);
                                           ERROR IS CATCHED BUT THROWS HERE AGAIN ---^

        return;
      }
    } else if (service.isHeatingZone) {
      try {
        const properties = (await this.platform.fibaroClient.getHeatingZone(IDs[0])).body.properties;
                                           SO THIS IDs[0] MUST TROW AN ERROR ---^
        if (!Object.prototype.hasOwnProperty.call(properties, 'currentTemperature')) {
          this.platform.log('No value for Temperature.', '');
          return;
        }
        characteristic.updateValue(properties.currentTemperature);
      } catch (e) {
        this.platform.log('There was a problem getting value from: ', `${service.IDs[0]} - Err: ${e}`);
                                           ERROR IS CATCHED BUT THROWS HERE AGAIN ---^
        return
      }
    } else {
      const value = properties.value;
      characteristic.updateValue(value);
    }
  }

```

So IMHO the problem is not a particular device, but initalizing the function call with `IDs = null`.

**Comment by mkz212** [comment]
@Krillle 

Thank you - this is a valuable tip. I have looking around it (this part of code) many times, but not having these specific devices that cause the problem is hard to test.

Tell me, have you been able to determine (e.g. by excluding individual devices as @ilcato previously advised) which device is causing the problem?


@Krillle @ilcato 
The "Cannot read properties of undefined (reading '0')" error occurs when accessing an undefined value at index 0. To solve the error, initialize the variable to the correct data type, e.g. an array or a string, before accessing the index.
https://bobbyhadz.com/blog/javascript-cannot-read-property-0-of-undefined

**Comment by Krillle** [comment]
Yes, it is the heating zone of device 131. But the issue is in the code itself:

The function `getCurrentTemperature(characteristic, service, IDs, properties)` is called with`IDs = null`, while `IDs` is crucial to read the current temperature of the heating zones:

`const properties = (await this.platform.fibaroClient.getHeatingZone(IDs[0])).body.properties;`
([/var/lib/homebridge/node_modules/homebridge-fibaro-home-center/src/getFunctions.ts:190:81](https://github.com/ilcato/homebridge-fibaro-home-center/blob/1671cbd266e7ca6806a009dd905e95e30e8b707f/src/getFunctions.ts#L190))

This will never deliver a result, and always throw an error:
```
IDs = null
getHeatingZone(IDs[0]))
```

The function is called in at Poller.manageValue, where it should hand over the array of devices (in my case `[131]`, not `null` as a constant.

`getFunction.function.call(this.platform.getFunctions, subscription.characteristic, subscription.service, null, change);
`
([/var/lib/homebridge/node_modules/homebridge-fibaro-home-center/src/pollerupdate.ts:110:34](https://github.com/ilcato/homebridge-fibaro-home-center/blob/1671cbd266e7ca6806a009dd905e95e30e8b707f/src/pollerupdate.ts#L110))

**Comment by Krillle** [comment]
@mkz212 @ilcato  can it be the issue lies in the logic itself: A heating zone has no current temperature, only the single devices in the heating zone.
So trying to call `getCurrentTemperature` for a heating zone doesn't make sense. Called with `IDs  = null` as a way to deal with it must fail.

**Comment by mkz212** [comment]
@Krillle Thanks again for all your help.

We have already corrected the code so that it describes more accurately which part of the code refers to. It will be in the next release (most likely still beta).

If you have HC3 then you have Climate Zone - Heating zone is in the old API for HC2 and HCL

You need to know that IDs is not a device id - this parameter is responsible for something completely different.

The most important thing, however, is that @ilcato probably already knows what the problem is and how he finds time will improve it.

**Comment by ilcato** [comment]
@Krillle @mkz212 
For climate zoned the ID is the ID of the panel containing the devices, If you have 3 climate zone the first is ID=1 the second 2 and so on.
In my original tests the climate zones where not reported in the refresh API so the getTemperature was not called from the poller mechanism.
It seems that this is not the case and for some reason we are receiving the change.
The IDs array is easily built from the subtype property.
I made this change in homebridge-fibaro-home-center@1.5.0-beta.5
@Krillle can you check it?

It could also be a clash in the ID space, maybe @Krillle you have a phisical device with a low id that collapse with the panel ids. Which is the lowest device id you have and which is the number of climate zone you have? If this is the case I need to manage this case.

**Comment by Krillle** [comment]
@ilcato  as mentioned above, the issue is in the code itself in my opinion. I think the problem lies here:
at Poller.manageValue (/var/lib/homebridge/node_modules/homebridge-fibaro-home-center/src/pollerupdate.ts:110:34):
```
getFunction.function.call(this.platform.getFunctions, subscription.characteristic, subscription.service, null, change);
                                                                FUNCTION IS CALLED WITH PARAMETER null ---^
```
at GetFunctions.getCurrentTemperature (/var/lib/homebridge/node_modules/homebridge-fibaro-home-center/src/getFunctions.ts:197:85):
 ```
 async getCurrentTemperature(characteristic, service, IDs, properties) {
                                                       ^--- IDs IS INITIALIZED WITH null
    if (service.isClimateZone) {
      try {
        const properties = (await this.platform.fibaroClient.getClimateZone(IDs[0])).body.properties;
                                           SO THIS IDs[0] MUST TROW AN ERROR ---^
        if (!Object.prototype.hasOwnProperty.call(properties, 'currentTemperatureHeating')) {
          this.platform.log('No value for Temperature.', '');
          return;
        }
        characteristic.updateValue(properties.currentTemperatureHeating);
      } catch (e) {
        this.platform.log('There was a problem getting value from: ', `${service.IDs[0]} - Err: ${e}`);
                                           ERROR IS CATCHED BUT THROWS HERE AGAIN ---^

        return;
      }
    } else if (service.isHeatingZone) {
      try {
        const properties = (await this.platform.fibaroClient.getHeatingZone(IDs[0])).body.properties;
                                           SO THIS IDs[0] MUST TROW AN ERROR ---^
        if (!Object.prototype.hasOwnProperty.call(properties, 'currentTemperature')) {
          this.platform.log('No value for Temperature.', '');
          return;
        }
        characteristic.updateValue(properties.currentTemperature);
      } catch (e) {
        this.platform.log('There was a problem getting value from: ', `${service.IDs[0]} - Err: ${e}`);
                                           ERROR IS CATCHED BUT THROWS HERE AGAIN ---^
        return
      }
    } else {
      const value = properties.value;
      characteristic.updateValue(value);
    }
  }

```

So IMHO the problem is not a particular device, but initalizing the function call with `IDs = null`.

**Comment by ilcato** [comment]
> @ilcato as mentioned above, the issue is in the code itself in my opinion. I think the problem lies here: at Poller.manageValue (/var/lib/homebridge/node_modules/homebridge-fibaro-home-center/src/pollerupdate.ts:110:34):
> 
> ```
> getFunction.function.call(this.platform.getFunctions, subscription.characteristic, subscription.service, null, change);
>                                                                 FUNCTION IS CALLED WITH PARAMETER null ---^
> ```
> 
> at GetFunctions.getCurrentTemperature (/var/lib/homebridge/node_modules/homebridge-fibaro-home-center/src/getFunctions.ts:197:85):
> 
> ```
> async getCurrentTemperature(characteristic, service, IDs, properties) {
>                                                       ^--- IDs IS INITIALIZED WITH null
>    if (service.isClimateZone) {
>      try {
>        const properties = (await this.platform.fibaroClient.getClimateZone(IDs[0])).body.properties;
>                                           SO THIS IDs[0] MUST TROW AN ERROR ---^
>        if (!Object.prototype.hasOwnProperty.call(properties, 'currentTemperatureHeating')) {
>          this.platform.log('No value for Temperature.', '');
>          return;
>        }
>        characteristic.updateValue(properties.currentTemperatureHeating);
>      } catch (e) {
>        this.platform.log('There was a problem getting value from: ', `${service.IDs[0]} - Err: ${e}`);
>                                           ERROR IS CATCHED BUT THROWS HERE AGAIN ---^
> 
>        return;
>      }
>    } else if (service.isHeatingZone) {
>      try {
>        const properties = (await this.platform.fibaroClient.getHeatingZone(IDs[0])).body.properties;
>                                           SO THIS IDs[0] MUST TROW AN ERROR ---^
>        if (!Object.prototype.hasOwnProperty.call(properties, 'currentTemperature')) {
>          this.platform.log('No value for Temperature.', '');
>          return;
>        }
>        characteristic.updateValue(properties.currentTemperature);
>      } catch (e) {
>        this.platform.log('There was a problem getting value from: ', `${service.IDs[0]} - Err: ${e}`);
>                                           ERROR IS CATCHED BUT THROWS HERE AGAIN ---^
>        return
>      }
>    } else {
>      const value = properties.value;
>      characteristic.updateValue(value);
>    }
>  }
> ```
> 
> So IMHO the problem is not a particular device, but initalizing the function call with `IDs = null`.

@Krillle in the beta5 the IDs is populated in the caller and passed to the function instead of null, see lines 110 and 111 of pollerUpdate.ts.

**Comment by Krillle** [comment]
> It could also be a clash in the ID space, maybe @Krillle you have a phisical device with a low id that collapse with the panel ids. Which is the lowest device id you have and which is the number of climate zone you have? If this is the case I need to manage this case.

I have a HC2 with 4 Heating Zones.

The system devices like controller ad rooms start with low ID from 1:
`{"id":1,"name":"zwave","roomID":0,"type":"com.fibaro.zwavePrimaryController"`
`{"id":2,"name":"xxxx","roomID":0,"type":"HC_user","baseType":"com.fibaro.voipUser"`
` {"id":3,"name":"YR Weather","roomID":0,"type":"com.fibaro.yrWeather","ba`
`{"id":4,"name":"4","roomID":0,"type":"com.fibaro.zwaveDevice","baseType":"com.fibaro.device","enabled":true`

Possible edge case: The heating zone wich causes the issue has 4 rooms, but only one room as a heating device. Three rooms don't.

The problematic heating zone is the only zone which appears in the log files:
```
[15/11/2023, 16:00:22] [FibaroHC] There was a problem getting value from:  Err: TypeError: Cannot read properties of null (reading '0') // I avoided the crash locally
[15/11/2023, 16:00:22] [FibaroHC] Wohnbereich [131]: 23.0 C
[15/11/2023, 16:00:22] [FibaroHC] Wohnbereich [131]: 0 
[15/11/2023, 16:00:22] [FibaroHC] Temperatur 129.0.1 [131]: 24.6 C
```

**Comment by ilcato** [comment]
@Krillle are you able to test beta5? In my setup there's no problems

**Comment by Krillle** [comment]
@ilcato just installed and monitoring the log files

**Comment by Krillle** [comment]
@ilcato  Yes, I see a thermostat for all four heating zones. And they set the temperature as manually set temperature for 2 hours as expected.

The current temperature shown in HomeApp is not related to the thermostat in heating zone, e.g.
Heating zone "Bedrooms" has just one room "Bedroom" with one HeatIt Thermostat:
Setpoint (target temperature) ist shown correctly in Home App.
Current temperature shown on the device on wall is 23.5°, while Home App shows 22°.

Heating zone "Livingarea" has 4 rooms, while only in "Livingroom" there is one HeatIt Thermostat:
Setpoint (target temperature) ist shown correctly in Home App.
Current temperature shown on the device on wall is 24.5°, while Home App shows 23°.

**Comment by Krillle** [comment]
@ilcato now a second heating zone showed up in the logs:

```
[15/11/2023, 16:24:21] [FibaroHC] Temperatur 133.0.1 [135]: 29.7 C
[15/11/2023, 16:25:02] [FibaroHC] Bäder [130]: 23.0 C
[15/11/2023, 16:25:02] [FibaroHC] Bäder [130]: 0 
[15/11/2023, 16:25:12] [FibaroHC] Helligkeit 12.0.2 [15]: 147 lux
[15/11/2023, 16:25:52] [FibaroHC] Temperatur 145.0.1 [147]: 22.5 C
[15/11/2023, 16:26:03] [FibaroHC] power is not a number. 
[15/11/2023, 16:26:03] [FibaroHC] power is not a number. 
[15/11/2023, 16:26:03] [FibaroHC] power is not a number. 
[15/11/2023, 16:29:38] [FibaroHC] power is not a number. 
[15/11/2023, 16:29:38] [FibaroHC] power is not a number. 
[15/11/2023, 16:29:38] [FibaroHC] power is not a number. 
[15/11/2023, 16:30:16] [FibaroHC] Helligkeit 12.0.2 [15]: 0 lux
[15/11/2023, 16:30:31] [FibaroHC] Wohnbereich [131]: 23.0 C
[15/11/2023, 16:30:31] [FibaroHC] Wohnbereich [131]: 0 
[15/11/2023, 16:30:31] [FibaroHC] Temperatur 129.0.1 [131]: 24.4 C
[15/11/2023, 16:31:00] [FibaroHC] Temperatur 44.0.1 [46]: 23.7 C
[15/11/2023, 16:31:37] [FibaroHC] Temperatur 141.0.1 [143]: 23.6 C
[15/11/2023, 16:33:02] [FibaroHC] Temperatur 133.0.1 [135]: 29.5 C
[15/11/2023, 16:34:25] [FibaroHC] power is not a number. 
[15/11/2023, 16:34:25] [FibaroHC] power is not a number. 
[15/11/2023, 16:34:25] [FibaroHC] power is not a number. 
[15/11/2023, 16:34:31] [FibaroHC] Temperatur 145.0.1 [147]: 21.9 C
[15/11/2023, 16:35:26] [FibaroHC] Temperatur 171.0.1 [173]: 25.2 C
[15/11/2023, 16:35:56] [FibaroHC] Temperatur 141.0.1 [143]: 23.7 C
[15/11/2023, 16:37:19] [FibaroHC] Temperatur 133.0.1 [135]: 29.3 C
[15/11/2023, 16:38:34] [FibaroHC] Temperatur 297.0.2 [300]: 24.4 C
[15/11/2023, 16:38:51] [FibaroHC] Temperatur 145.0.1 [147]: 21.1 C
[15/11/2023, 16:39:11] [FibaroHC] Wohnbereich [131]: 23.0 C
[15/11/2023, 16:39:11] [FibaroHC] Wohnbereich [131]: 0 
[15/11/2023, 16:39:11] [FibaroHC] Temperatur 129.0.1 [131]: 24.5 C
```

**Comment by Krillle** [comment]
@ilcato with homebridge-fibaro-home-center@1.5.0-beta.5 I have all devices which were respecting the device type in Home Center double again with the wrong type, e.g what is set as a switch is now a socket as well

**Comment by mkz212** [comment]
@Krillle see Advanced Control in plugin settings

**Comment by mkz212** [comment]
@Krillle Did the fix that pushed @ilcato out in one of the last betas solve your problem? If so, please close the issue.

**Comment by Krillle** [comment]
Fixed with v1.5.0-beta.5
Thank you for good work.
